### ISSUE TITLE
ExecuteScalarAsync not working when trying to bring back COUNT

### ISSUE THREAD

**Issue by CodeLiftSleep** [praeclarum/sqlite-net#605]
conn.ExecuteScalarAsync<int>("SELECT COUNT(*) FROM DraftPlayers", null).ContinueWith((t) => {
                    Draftpages = t.Result;
                });
I know the connection path works because right above it in code I load a bunch of players to the game.

The result comes back as 0 and never continues inside the ContinueWith statment, however the same statement inside the DB itself gives 3000.

**Comment by praeclarum** [comment]
I will have to write a unit test for this. Thanks for reporting!

Have you tried the latest nuget (there is a v1.5 in prerelease)? This bug sounds familiar...

**Comment by CodeLiftSleep** [comment]
No I have not tried with 1.5...I'll check it out and report back

**Comment by CodeLiftSleep** [comment]
I stepped through the code and it gives me a `SystemAggregate.Exception: one or more errors occured ` 
  when it goes to t.Result--it does go in there, but it throws an error...

**Comment by CodeLiftSleep** [comment]
Tried running it using ExecuteScalar and this time got a real meaningful error:

`Exception thrown: Object reference not set to an instance of an object
  at SQLite.SQLiteConnection.CreateCommand (System.String cmdText, System.Object[] ps) [0x00044] in C:\Users\MNasty\Documents\WPFFootball2017\Assets\SQLite-Net\SQLite.cs:815 
  at SQLite.SQLiteConnection.ExecuteScalar[T] (System.String query, System.Object[] args) [0x00001] in`

**Comment by CodeLiftSleep** [comment]
OK---Solved it.

Actually I just took read the description of the function being called and realized you have an error in your code example.  

The example you give is: db.ExecuteScalarAsync<int>(SELECT Count(*) FROM xxxx", null);

This is wrong.  The second argument is a parmeter array which is always optional.  So passing in null actually passes in null as a parameter.  When I realized this and removed null as the second argument and simply put:

db.ExecuteScalarAsync<int>(SELECT Count(*) FROM xxxx");  it worked without any issues.  Might want to correct your example.
